### PR TITLE
Fix!: preserve ascending order of sorting when present

### LINKED ISSUE #2253: sort order is not rendered for duckdb
**Fully reproducible code snippet**

DuckDB has a setting `default_order`, which can be set to `'DESC'`, and results in unspecified sort order in `ORDER BY` queries being ordered according to `default_order`.

sqlglot seems to assume (reasonably so, I think) that `ORDER BY x ASC` is equivalent to `ORDER BY x`, which is an invalid assumption for duckdb given the presence of `default_order`:

```
In [4]: import sqlglot as sg

In [5]: sg.__version__
Out[5]: '18.5.1'

In [6]: sg.parse_one("select i from range(5) _ (i) order by i asc", read="duckdb").sql("duckdb")
Out[6]: 'SELECT i FROM RANGE(5) AS _(i) ORDER BY i'

In [7]: !duckdb
D set default_order = 'DESC';
D select i from range(5) _ (i) order by i;
┌───────┐
│   i   │
│ int64 │
├───────┤
│     4 │
│     3 │
│     2 │
│     1 │
│     0 │
└───────┘
```

**Official Documentation**

- https://duckdb.org/docs/sql/query_syntax/orderby.html

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Fixes #2253

### CODE REVIEW

File: sqlglot/generator.py [hunk 1]
```diff
@@ -1665,7 +1670,7 @@ def ordered_sql(self, expression: exp.Ordered) -> str:
         nulls_are_small = self.NULL_ORDERING == "nulls_are_small"
         nulls_are_last = self.NULL_ORDERING == "nulls_are_last"
 
-        sort_order = " DESC" if desc else ""
+        sort_order = " DESC" if desc else f"{self.default_order}"
```
Comment (tobymao) [inline sqlglot/generator.py:1673]: self.default_order, don't need the quotes

File: sqlglot/generator.py [hunk 2]
```diff
@@ -46,6 +46,8 @@ class Generator:
             The default is on the smaller end because the length only represents a segment and not the true
             line length.
             Default: 80
+        default_order: The default order to use in the ORDER BY clause if ASC / DESC are omitted.
```
Comment (tobymao) [inline sqlglot/generator.py:49]: i don't think this should be passed in a string, just a boolean

Comment (GeorgeSittas) [reply sqlglot/generator.py:49]: sounds good with me, will update it so it expects an optional boolean and `None` will stand for `""`.

Comment (GeorgeSittas) [reply sqlglot/generator.py:49]: yea I was thinking something like `sort_asc` or something

Comment (cpcloud) [reply sqlglot/generator.py:49]: Actually, I don't think I under what `True` or `False` would mean here.

Does `True` mean "always render the order" and `False` mean "render only if `DESC`?

Comment (GeorgeSittas) [reply sqlglot/generator.py:49]: If the arg was named `default_order_is_asc`, then `True` means `ASC` is printed in the "default" `ORDER BY`, `False` means `DESC` is printed and `None` means it's left as is, i.e `ORDER BY a`.

I'm actually taking a closer look again to see if I can just preserve `ASC` / `DESC` as you suggested in the issue. I tried to do this initially but we have a lot of test failures and I need to verify it's a safe change to make.

Comment (GeorgeSittas) [reply sqlglot/generator.py]: https://github.com/tobymao/sqlglot/pull/2256/commits/198099d80a671684680bbd89039a0c5bfefe998b

File: sqlglot/parser.py [hunk 3]
```diff
@@ -2944,20 +2944,20 @@ def _parse_sort(self, exp_class: t.Type[E], token: TokenType) -> t.Optional[E]:
 
     def _parse_ordered(self) -> exp.Ordered:
         this = self._parse_conjunction()
-        self._match(TokenType.ASC)
 
-        is_desc = self._match(TokenType.DESC)
+        asc = self._match(TokenType.ASC)
+        desc = self._match(TokenType.DESC) or (asc and False)
```
Comment (tobymao) [inline sqlglot/parser.py:2949]: what and False?

File: tests/fixtures/optimizer/tpc-ds/tpc-ds.sql [hunk 4]
```diff
@@ -4529,7 +4529,7 @@ SELECT
   "item"."i_category" AS "i_category",
   "item"."i_class" AS "i_class",
   GROUPING("item"."i_category") + GROUPING("item"."i_class") AS "lochierarchy",
-  RANK() OVER (PARTITION BY GROUPING("item"."i_category") + GROUPING("item"."i_class"), CASE WHEN GROUPING("item"."i_class") = 0 THEN "item"."i_category" END ORDER BY SUM("store_sales"."ss_net_profit") / SUM("store_sales"."ss_ext_sales_price")) AS "rank_within_parent"
+  RANK() OVER (PARTITION BY GROUPING("item"."i_category") + GROUPING("item"."i_class"), CASE WHEN GROUPING("item"."i_class") = 0 THEN "item"."i_category" END ORDER BY SUM("store_sales"."ss_net_profit") / SUM("store_sales"."ss_ext_sales_price") ASC) AS "rank_within_parent"
```
Comment (tobymao) [inline tests/fixtures/optimizer/tpc-ds/tpc-ds.sql:4532]: maybe the the optimizer canonicalize should get rid of asc

File: sqlglot/parser.py (hunk 3 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py:2949]: It could be rewritten as `None if asc is None else False`

File: tests/fixtures/optimizer/tpc-ds/tpc-ds.sql (hunk 4 above)
Comment (GeorgeSittas) [reply tests/fixtures/optimizer/tpc-ds/tpc-ds.sql:4532]: Why should we get rid of it?

File: sqlglot/parser.py (hunk 3 above)
Comment (tobymao) [reply sqlglot/parser.py:2949]: ok, because you want it to be none? that's fine

File: tests/fixtures/optimizer/tpc-ds/tpc-ds.sql (hunk 4 above)
Comment (tobymao) [reply tests/fixtures/optimizer/tpc-ds/tpc-ds.sql:4532]: because order by x == order by x asc

Comment (GeorgeSittas) [reply tests/fixtures/optimizer/tpc-ds/tpc-ds.sql:4532]: should we do this in a separate PR?

File: sqlglot/parser.py (hunk 3 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py:2949]: yes that was the intent

File: tests/fixtures/optimizer/tpc-ds/tpc-ds.sql (hunk 4 above)
Comment (tobymao) [reply tests/fixtures/optimizer/tpc-ds/tpc-ds.sql:4532]: no, just do it here, it affects these tests

Comment (GeorgeSittas) [reply tests/fixtures/optimizer/tpc-ds/tpc-ds.sql]: done

**Comment by cpcloud** [comment]
Just to verify, the ASC Will be removed in an optimization pass?

Won't that reintroduce the bug if I send in a query with ASC, optimize it, and then send that optimized SQL to duckdb with a different default sort order?

**Comment by cpcloud** [comment]
Is there an issue with keeping ASC around? Are there databases for which that is a meaningful optimization?

**Comment by tobymao** [comment]
in general order by x == order by x asc, duckdb setting is an anomoly. we canonicalize to make sql equal. if you want to make it an option to turn that off you can, or remove canonicalization

**Comment by cpcloud** [comment]
Got it!

Wouldn't SQL generated with ASC still be canonical? Are there dialects where ASC isn't supported?

Happy to discuss, just not sure I'm following the purpose of removing the ASC. Is the idea to keep the generated code smaller?

**Comment by cpcloud** [comment]
If it's just a weird DuckDB setting then I guess I can live with it. Hopefully people aren't altering that setting too much 😬😬

**Pull request merged**